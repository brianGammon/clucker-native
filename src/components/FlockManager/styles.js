/* @flow */
import { StyleSheet } from 'react-native';
import variables from '../../styles/variables';

export default StyleSheet.create({
  noFlocksMessage: {
    marginVertical: 10,
    flexDirection: 'row',
    backgroundColor: variables.cluckerOrangeLight,
    borderRadius: 20,
  },
  noFlocksMessageText: {
    flex: 1,
  },
  image: {
    width: 50,
    height: 50,
  },
});
