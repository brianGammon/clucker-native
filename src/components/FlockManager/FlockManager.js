/* @flow */
import React from 'react';
import { Image } from 'react-native';
import { View, Text } from 'native-base';
import FlockSelector from '../FlockSelector';
import AddFlock from '../AddFlock';
import JoinFlock from '../JoinFlock';
import styles from './styles';

type Props = {
  hasFlocks: boolean,
};

const FlockManager = ({ hasFlocks }: Props) => (
  <View>
    {!hasFlocks && (
      <View padder style={styles.noFlocksMessage}>
        <Image
          source={require('../../assets/chicken-sm.png')}
          style={styles.image}
        />
        <Text style={styles.noFlocksMessageText}>
          To start tracking eggs, either create a new flock or join an existing
          flock below.
        </Text>
      </View>
    )}
    {hasFlocks && <FlockSelector />}
    <AddFlock />
    <JoinFlock />
  </View>
);

export default FlockManager;
