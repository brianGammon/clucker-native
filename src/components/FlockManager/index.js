/* @flow */
import FlockManager from './FlockManager';

export default FlockManager;
