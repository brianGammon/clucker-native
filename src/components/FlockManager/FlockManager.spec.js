import React from 'react';
import FlockManager from '.';

describe('FlockManager', () => {
  test('renders with flocks', () => {
    const wrapper = shallow(<FlockManager hasFlocks />);
    expect(wrapper).toMatchSnapshot();
  });

  test('renders without flocks', () => {
    const wrapper = shallow(<FlockManager hasFlocks={false} />);
    expect(wrapper).toMatchSnapshot();
  });
});
