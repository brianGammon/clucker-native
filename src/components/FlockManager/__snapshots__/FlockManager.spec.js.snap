// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlockManager renders with flocks 1`] = `
<Styled(ViewNB)>
  <Connect(FlockSelector) />
  <Connect(AddFlock) />
  <Connect(JoinFlock) />
</Styled(ViewNB)>
`;

exports[`FlockManager renders without flocks 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)
    padder={true}
    style={
      Object {
        "backgroundColor": "rgb(254, 206, 137)",
        "borderRadius": 20,
        "flexDirection": "row",
        "marginVertical": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/chicken-sm.png",
        }
      }
      style={
        Object {
          "height": 50,
          "width": 50,
        }
      }
    />
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      To start tracking eggs, either create a new flock or join an existing flock below.
    </Styled(Text)>
  </Styled(ViewNB)>
  <Connect(AddFlock) />
  <Connect(JoinFlock) />
</Styled(ViewNB)>
`;
